src/UI/TextField.elm

```diff
@@ -159,6 +159,29 @@ type TextFieldWidth
 -- Options
 
 
+{-| Irreversibly converts the input to static field. Given True to a condition.
+
+    TextField.newPassword Msg.OnPasswordChanged
+        "New password"
+        model.value
+        |> TextField.toStaticIf
+            model.hasSentForm
+
+-}
+toStaticIf : Bool -> TextField msg -> TextField msg
```

Comment:
I would keep using `with*`, we do have something similar in the UI.Button, I would call it `withStaticIf`.

That being said, I would also rename the term used here, I'd prefer disabled instead of static, I think `disabled` is widely used and it's more natural in my opinion in the frontend world.

Reply:
The thing about `to` is: This is an irreversible change to the TextField.

> I think disabled is widely used and it's more natural in my opinion in the frontend world.

Static fields can't be "reenabled". They use a different construction process and even aren't real input fields. They're common elements mimicing the input appearence.

Another solution would be to eradicate the current way of creating a static field and make the message an optional/maybe field for all constructors.

Reply:
I didnt get why it is irreversible, this function receives a boolean, AFAIK if the boolean it's true the field starts to work again right? in the create a user story, if the backend returns an error, we enabled the field again right?

Reply:
> if the boolean it's true the field starts to work again right?

Nope, this flow would happen:
* construct (with `currentPassword`, `username` , etc) -> TextField with _Msg_ and with _Type_ -> `withStaticIf True` -> TextField with _Type_ and without _Msg_
* construct (with `static`) -> TextField without _Msg_ and without _Type_ -> `withStaticIf False` -> **missing** _Msg_ and _Type_

Reminding you that a disabled/static doesn't have a message cause we don't want to creat a "Msg.NoOp" for a static field..

A better API would have all the constructors with `Maybe Msg`, where being Nothing means it's disabled. This way we could create passwords and multilines "disabled" fields.

(I'm not enforcing either of this two solutions, I'm really open to discussion about this)

Reply:
I think we might be stepping our own toes here. I don't see reason not to use the "real" disabled behavior. Turning it on and off would not longer be a problem in that case and it would be much simpler too.

Reply:
Moreover, I think we should lean into the language we're familiar with when it makes sense. I didn't immediately get what the whole "static" thing was about, so calling it either disabled or read only would be better in that sense